src/lib.rs

```diff
@@ -145,10 +231,11 @@ impl TypeMap {
     pub fn get<T>(&self) -> Option<&T::Value>
     where
         T: TypeMapKey,
+        T::Value: IntoBox<S>,
     {
         self.0
             .get(&TypeId::of::<T>())
-            .and_then(|b| b.downcast_ref::<T::Value>())
+            .map(|b| unsafe { &*((&**b) as *const _ as *const T::Value) })
```

Comment:
This should be safe... I think. I've tried running the code through Miri, which doesn't complain so it should be okay?